Extension dbt documentation - Add ability to scrap code snippets

b-per opened the pull request:
## Description

<!-- A summary of your change. If you add a new extension or command, explain what it does. -->
I have updated my dbt-documentation extension to:
- scratch code snippets from the doc page selected and show them to users
- allow people to copy those code snippets
- show better description in the list based on the hierarchy values returned by Algolia
- use the latest raycast API


## Screencast

<!-- If you add a new extension or command, include a screencast (or screenshot for straightforward changes). A good screencast will make the review much faster - especially if your extension requires registration in other services.  -->

Ability for users to "Show Code Snippets" in a given documentation page
<img width="1000" alt="dbt-documentation 2022-07-21 at 10 58 56" src="https://user-images.githubusercontent.com/8754100/180182506-4694ed70-d862-465b-ac02-c1e5c1bb6287.png">

Ability for users to search and copy those snippets
<img width="1000" alt="dbt-documentation 2022-07-21 at 11 09 34" src="https://user-images.githubusercontent.com/8754100/180182057-3eba957f-9470-4656-9e6d-217001c09eb3.png">


## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

danielsequeira commented:
Hey @b-per,

Just two small suggestions from my side:

1. When documentation entry has no code snippets, you could not display the right sidebar and only display the no results text and icon in the middle of the screen:
<img width="1000" alt="dbt-documentation 2022-07-21 at 17 00 23" src="https://user-images.githubusercontent.com/12091319/180246997-bfa7c656-9b7c-468c-9ca0-d1a4c795140b.png">

2. Why don't you add an icon to the Show Code Snippets action?
<img width="1000" alt="dbt-documentation 2022-07-21 at 17 01 12" src="https://user-images.githubusercontent.com/12091319/180247143-33e09d55-1185-43b5-be11-94541cc71bae.png">

b-per commented:
Thanks for the review @danielsequeira !

- I have updated the code to follow your recommended approach to 1.
![dbt-documentation 2022-07-21 at 19 39 45](https://user-images.githubusercontent.com/8754100/180278392-63e99427-8e66-4ed4-bb5b-9b29158ce429.png)

- There was no real reason for me not to have an icon for 2. so I added one. I have updated the screenshot for the extension as well
<img width="1000" alt="dbt-documentation 2022-07-21 at 19 34 14" src="https://user-images.githubusercontent.com/8754100/180278466-4f76b83e-428f-4ed4-8d84-54fbaffa995a.png">
